app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/FahrplanFragment.kt

```diff
@@ -430,7 +439,12 @@ class FahrplanFragment : Fragment(), MenuProvider {
                     )
                 }
             }
-            columnsLayout.addView(roomColumnView)
+
+            if (columnsLayout.size > roomIndex) {
+                columnsLayout.removeViewAt(roomIndex)
+            }
+            columnsLayout.addView(roomColumnView, roomIndex)
+            roomDataCache[roomData.roomName] = roomData
```

Comment:
We also parse the `guid` or a `room` from the XML. It is available as `roomIdentifier` in the `Session` model. `SessionsTransformer` does not pass it through into `RoomData` but we could add it.